Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -337,20 +617,29 @@ class GeneralForceSubsystemRep : public ForceSubsystem::Guts {
         const Array_<bool>& enabled = Value<Array_<bool> >::downcast
             (getDiscreteVariable(s, forceEnabledIndex));
         for (int i = 0; i < (int) forces.size(); ++i)
-            if (enabled[i]) 
+            if (enabled[i])
                 forces[i]->getImpl().calcDecorativeGeometryAndAppend(s,stage,geom);
         return 0;
     }
 
 private:
     Array_<Force*>                  forces;
+
+    // For parallel calculation of forces.
+    ReferencePtr<ParallelExecutor>               calcForcesExecutor;
+    ReferencePtr<CalcForcesTask>                 calcForcesTask;
```

Comment:
I don't think this is correct use of ReferencePtr. See documentation -- ReferencePtr cannot be used for any heap space that you own. Could use std::unique_ptr for that, or just go back to raw pointers here since the (ugly) explicit memory management code was already there.